server/src/test/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetServiceTests.java

```diff
@@ -248,6 +249,7 @@ public void testPrepareIndexForPeerRecovery() throws Exception {
         }
         assertThat(replica.recoveryState().getTranslog().recoveredOperations(), equalTo(0));
         assertThat(replica.getLastKnownGlobalCheckpoint(), equalTo(UNASSIGNED_SEQ_NO));
+        assertThat(replica.recoveryState().getStage(), equalTo(RecoveryState.Stage.TRANSLOG));
```

Comment:
This assertion will fail without the fix. And by always using a globalCheckpoint < maxSeqNo, the test will always fail and not occasionally only (like it does on the original commit).